### ISSUE TITLE
[Bug]: pessimistic transaction and rc mode： mo crashed

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#10392]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):d14c80ecb98762ef039d726d63fb0dd5b418ceab
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

1cn pessmistic rc mode ,run bvt case , first cases failed ,then mo crashed
![image](https://github.com/matrixorigin/matrixone/assets/113406637/815d2ca6-a9ef-466d-9f3b-ebb54e561a04)
![image](https://github.com/matrixorigin/matrixone/assets/113406637/b857e7df-6008-45ce-a37a-3f055f06bf05)

mo crashed,mo log:
2023/07/03 16:41:40.103037 +0800 ERROR frontend/util.go:522 data out of range: data type int64, value '9223372036854775808' {"session_info": "connectionId 9|127.0.0.1:53554|{account sys:dump:moadmin -- 0:1:0}|goRoutineId 27405|6d225166-197d-11ee-a730-96f0c1a980fd"}
2023/07/03 16:41:40.103125 +0800 ERROR frontend/util.go:522 query trace status {"connection_id": 9, "statement": "insert into t1 values (9223372036854775808,'xefrsdfgds')", "status": "fail", "error": "data out of range: data type int64, value '9223372036854775808'", "span": {"trace_id": "6dee0a2c-197d-11ee-a730-96f0c1a980fd", "kind": "statement"}, "session_info": "connectionId 9|127.0.0.1:53554|{account sys:dump:moadmin -- 0:1:0}|goRoutineId 27405|6d225166-197d-11ee-a730-96f0c1a980fd"}
2023/07/03 16:41:40.110569 +0800 ERROR incrservice/column_cache.go:154 error: data out of range: data type bigint, value 9223372036854775808 {"span": {"trace_id": "6deedaec-197d-11ee-a730-96f0c1a980fd", "kind": "statement"}}
2023/07/03 16:41:40.111108 +0800 ERROR colexec/receiver_operator.go:93 children pipeline closed unexpectedly
2023/07/03 16:41:40.111585 +0800 ERROR frontend/util.go:522 data out of range: data type bigint, value 9223372036854775808 {"session_info": "connectionId 9|127.0.0.1:53554|{account sys:dump:moadmin -- 0:1:0}|goRoutineId 27405|6d225166-197d-11ee-a730-96f0c1a980fd"}
2023/07/03 16:41:40.111681 +0800 ERROR frontend/util.go:522 query trace status {"connection_id": 9, "statement": "insert into t1(b) values ('eeeee')", "status": "fail", "error": "data out of range: data type bigint, value 9223372036854775808", "span": {"trace_id": "6deedaec-197d-11ee-a730-96f0c1a980fd", "kind": "statement"}, "session_info": "connectionId 9|127.0.0.1:53554|{account sys:dump:moadmin -- 0:1:0}|goRoutineId 27405|6d225166-197d-11ee-a730-96f0c1a980fd"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10192d574]

goroutine 28346 [running]:
github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).IsConst(...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/container/vector/vector.go:269
github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).IsConstNull(...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/container/vector/vector.go:331
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae/logtailreplay.EncodePrimaryKeyVector(0x14007cc1368?, 0x0?)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/vm/engine/disttae/logtailreplay/encode.go:27 +0x24
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae/logtailreplay.(*PartitionState).PrimaryKeysMayBeModified(0x0?, 0x0?, {0x0, 0x0, 0x0, 0x0, 0xf0, 0x35, 0x3a, 0x54, ...}, ...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/vm/engine/disttae/logtailreplay/primary_key_utils.go:31 +0x30
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*txnTable).PrimaryKeysMayBeModified(0x140075a2a80, {0x1029bea98, 0x1400bd207e0}, {0x0, 0x0, 0x0, 0x0, 0xf0, 0x35, 0x3a, ...}, ...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/vm/engine/disttae/txn_table.go:1799 +0x120
github.com/matrixorigin/matrixone/pkg/sql/colexec/lockop.hasNewVersionInRange(0x1400bb47180, 0x0?, {0x1029e7530, 0x1400b265b80}, 0x0?, {0x176e4eef543a35f0, 0x0, 0x0, {}, {0x0, ...}, ...}, ...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/colexec/lockop/lock_op.go:714 +0x40c
github.com/matrixorigin/matrixone/pkg/sql/colexec/lockop.doLock({0x1029bea98, 0x1400bd207e0}, 0x0?, {0x1029e7530, 0x1400b265b80}, 0x0?, 0x1400bb47180, 0x0?, {0x17, 0x0, ...}, ...)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/colexec/lockop/lock_op.go:417 +0x650
github.com/matrixorigin/matrixone/pkg/sql/colexec/lockop.LockTable({0x1029e7530, 0x1400b265b80}, 0x1400bb47180, 0x0?, {0x17, 0x0, 0x0, 0x0, 0x8, 0x40, ...})
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/colexec/lockop/lock_op.go:272 +0x240
github.com/matrixorigin/matrixone/pkg/sql/compile.lockTable({0x1029e7530, 0x1400b265b80}, 0x1400bb47180, {0x1029ffff8, 0x1400b3afc00})
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/compile/ddl.go:1800 +0xb4
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).DropTable(0x1400e392680, 0x1400a796480)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/compile/ddl.go:1247 +0x31c
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).run(0x1400a796480, 0x1400e392680)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/compile/compile.go:280 +0x378
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).runOnce.func1(0x0?)
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/compile/compile.go:369 +0x38
created by github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).runOnce
	/Users/heni/test-envir/dis_cn/matrixone/pkg/sql/compile/compile.go:368 +0xc0

mo log:
[log1.txt](https://github.com/matrixorigin/matrixone/files/11935081/log1.txt)

**Comment by reusee** [comment]
investigating

**Comment by zhangxu19830126** [comment]
fixed

**Comment by zhangxu19830126** [comment]
fixed by #10401

**Comment by heni02** [comment]
confirm, closed
commit id :4a65d4ba8aa8cc447af9167fab9ebb234970b376
